spec.bs

```diff
@@ -1856,55 +1875,95 @@ To <dfn>convert to an AuctionAd sequence</dfn> given a [=list=]-or-null |ads|:
+    limit is exceeded, the request must be divided into smaller pieces to comply with the length restriction.
+    However, note that a single request is always considered valid, regardless of whether its length exceeds
+    the auction's [=auction config/max trusted scoring signals url length=].
+
+    The network response has to be parsed to pull out the pieces relevant to each
+    [=evaluating a scoring script|evaluation of a scoring script=].
+
+    These requests may also begin before the script fetch, but requests cross-origin to the
+    script origin must not happen until [:Ad-Auction-Allow-Trusted-Scoring-Signals-From:] header on
+    the script is received, parsed, and determined to authorize such a fetch.
+  1. If |fullSignalsUrl| is not same [=same origin=] with |auctionConfig|'s
+     [=auction config/seller=], then:
+      1. Set |crossOriginTrustedScoringSignalsOrigin| to |fullSignalsUrl|'s [=url/origin=].
+      1. If |allowCrossOriginTrustedScoringSignalsFrom| does not [=list/contain=]
+        |crossOriginTrustedScoringSignalsOrigin|:
+        1. Set |crossOriginTrustedScoringSignalsOrigin| to null.
+        1. Set |fullSignalsUrl| to null.
+1. If |fullSignalsUrl| is not null:
+  1. Let |trustedScoringSignals| and |allTrustedScoringSignals| be null.
+  1. Set «|allTrustedScoringSignals|, ignored, |scoringDataVersion|» to the result of [=fetching
+    trusted signals=] with |fullSignalsUrl|, |auctionConfig|'s [=auction config/seller=], and false.
+  1. If |allTrustedScoringSignals| is an [=ordered map=]:
+    1. Set |trustedScoringSignals| to a new empty [=map=].
+    1. [=map/Set=] |trustedScoringSignals|["`renderURL`"] to a new empty [=map=].
+    1. If |allTrustedScoringSignals|["`renderURLs`"] [=map/exists=] and
+      |allTrustedScoringSignals|["`renderURLs`"][|renderURL|] [=map/exists=], then [=map/set=]
+      |trustedScoringSignals|["`renderURL`"][|renderURL|] to
+      |allTrustedScoringSignals|["`renderURLs`"][|renderURL|].
+    1. If |adComponentRenderURLs| is not [=list/empty=]:
+      1. Let |adComponentRenderURLsValue| be a new empty [=map=].
+      1. If |allTrustedScoringSignals|["`adComponentRenderURLs`"] [=map/exists=], [=set/for each=]
+        |adComponentRenderURL| in |adComponentRenderURLs|:
+        1. If |allTrustedScoringSignals|["`adComponentRenderURLs`"][|adComponentRenderURL|]
+          [=map/exists=], then [=map/set=] |adComponentRenderURLsValue|[|adComponentRenderURL|] to
+          |allTrustedScoringSignals|["`adComponentRenderURLs`"][|adComponentRenderURL|].
+      1. [=map/Set=] |trustedScoringSignals|["`adComponentRenderURLs`"] to |adComponentRenderURLsValue|.
+    1. If |crossOriginTrustedScoringSignalsOrigin| is null, set |sameOriginTrustedScoringSignals|
+      to |trustedScoringSignals|.
+    1. Otherwise:
+      1. Set |crossOriginTrustedScoringSignals| to a new [=ordered map=] whose [=map/keys=] are
+          [=strings=] and whose [=map/values=] are {{any}}.

```

Comment:
why does it need to be any? What can it be besides a map

Reply:
It used to be possible to get a null in there, but not any more... Anyway, I just made it a map w/o details since the value type is complicated.